stable/redis/README.md

```diff
@@ -45,6 +45,35 @@ $ helm delete my-release
 
 The command removes all the Kubernetes components associated with the chart and deletes the release.
 
+## Upgrading an existing Release to a new major version
+
+A major chart version change (like v1.2.3 -> v2.0.0) indicates that there is an
+incompatible breaking change needing manual actions.
+
+### 4.0.0
+
+This version removes the `chart` label from the `spec.selector.matchLabels`
+which is immutable since `StatefulSet apps/v1beta2`. It has been inadvertently
```

Comment:
@desaintmartin Aren't there two issues with this chart?
1. The `StatefulSet` is currently `apps/v1beta2` (issue #7680)
1. The `Deployment`s are currently `extensions/v1beta1` (issue #7726)

If I understand this correctly, would you please clarify in these upgrade notes, and link to each issue? (the reason is that while we have an understanding now of the best way to upgrade, there may be better methods discovered after this next version is released, but we can update those issues without changing the chart README, which required another version bump. Also we will have many charts that can all point to the same issue(s)).

Reply:
Hehe, thanks to your remark I found out something interesting. My solution described for extensions/v1beta1 (issue #7726) works well (notice in my solution that I add chart in explicit selector). But here, adding a new selector without chart makes k8s complain about selector != `spec.template.metadata.labels`. So I'm going to elaborate a documentation based around   https://github.com/helm/charts/issues/5657#issuecomment-422365447.

Reply:
Done.